app/javascript/components/Profile/UserList.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { Fragment } from 'react';
+import { Icon, Table, Button } from 'semantic-ui-react';
+import { inject, observer } from 'mobx-react';
+import { computed } from 'mobx';
+import UserStore, { ReloadState } from '../../stores/user_store';
+import User, { Role } from '../../models/User';
+import { deleteUser, updateUser } from '../../api/admin';
+import SessionStore from '../../stores/session_store';
+
+interface InjectedProps {
+  userStore: UserStore;
+  sessionStore: SessionStore;
+}
+
+@inject('userStore', 'sessionStore')
+@observer
+export default class UserList extends React.Component {
+  componentDidMount() {
+    this.injected.userStore.loadUsers();
```

Comment:
not sure about the flow (have to think about his a little more in depth), but here my concern:

shouldn't the userStore ensure that all of its users are initially loaded and stored in an observable public users variable? Every time we delete a user, the store's user list should be updated accordingly. (either be fetching all users from the back-end or by updating its user list).

Reply:
@simplay yes this seems to be more the mobx-way. I changed that and like it better this way :+1: